### ISSUE TITLE
Updating to 10.23.0 results in 'FirebaseCrashlytics/FirebaseCrashlytics-Swift.h' file not found

### ISSUE THREAD

**Issue by OscarFyl** [firebase/firebase-ios-sdk#12611]
### Description

![image](https://github.com/firebase/firebase-ios-sdk/assets/164495823/e4f6587e-5d06-4a03-857d-af088cb50f4c)

### Firebase Product(s)

Crashlytics

**Comment by rizafran** [comment]
Hi @OscarFyl, I tried to reproduce the issue, but I'm able to successfully run the app. Could you provide your `Podfile` and `Podfile.lock` file?

**Comment by macdrevx** [comment]
Same error as in #12605. Need more details to know if it's the same root cause.

**Comment by OscarFyl** [comment]
[Podfile.txt](https://github.com/firebase/firebase-ios-sdk/files/14729973/Podfile.txt)
[Podfile.lock.txt](https://github.com/firebase/firebase-ios-sdk/files/14729974/Podfile.lock.txt)
![image](https://github.com/firebase/firebase-ios-sdk/assets/164495823/b6cd780c-a592-405f-a3cb-52ee0164cdf6)
![image](https://github.com/firebase/firebase-ios-sdk/assets/164495823/8ed6be4e-77d2-4ea8-80b4-b040d7367cd4)
![image](https://github.com/firebase/firebase-ios-sdk/assets/164495823/7df5d421-6ebf-436b-b63c-4778b53efc00)
![image](https://github.com/firebase/firebase-ios-sdk/assets/164495823/3eb36b75-dd39-45a1-9df1-a9474a8856c8)
Hello
The file contains my podflile file and podflile.lock file, as well as screenshots of my project error reports. I was using version 10.20.0 before. When I upgraded to version 10.23.0, I would get the error 'FirebaseCrashlytics/FirebaseCrashlytics-Swift.h' file not found'. I tried the following:
1.clean project
2. Re-execute pod install and pod update
3. Cleaned up the DerivedData directory of xcode,
4. Exit Xcode and restart
None of the above four methods will work.
I can't find this file when I search for FirebaseCrashlytics-Swift in my project. I can't find it in the folder through show in finder. Where does this file exist?
My computer system is: Sonoma 14.4
Xcode version is: 15.3

**Comment by themiswang** [comment]
Hey @OscarFyl 
This is the same error as https://github.com/firebase/firebase-ios-sdk/issues/12605 mentioned, you need to uncomment `use_frameworks!` line. As @paulb777  mentioned in that issue `use_frameworks!` is required since Firebase 9.0. More details in the release notes - https://firebase.google.com/support/release-notes/ios.

**Comment by paul-lavoine** [comment]
Hello, I have the same problem.

But since the version 10.22.1 everything worked well.
I've never included use_frameworks! but I included use_modular_headers! instead and until the version 10.22.1 it was working,

Did something change ?

**Comment by themiswang** [comment]
Hey @paul-lavoine 

We added a swift module in Crashlytics for 10.23.0 https://github.com/firebase/firebase-ios-sdk/pull/12410. This is related to change in release note for 10.23.0 https://firebase.google.com/support/release-notes/ios#crashlytics

**Comment by dklt** [comment]
My project also got hit with this issue. 30+ external libraries, 10+ spm packages, mix of ObjectiveC + Swift codebase.

After applying the suggested fixes above, Xcode choke on more errors... 🤮 

reverting back to `10.22.0` until a stable release.

**Comment by VisionColion** [comment]
My project also encountered this problem. I saw that this has been closed, but how to solve it?

**Comment by paulb777** [comment]
Podfiles should include either `use_frameworks!` or `use_frameworks! :linkage => :static`

**Comment by haozhutw** [comment]
> Podfiles should include either `use_frameworks!` or `use_frameworks! :linkage => :static`

Hi @paulb777, `use_frameworks!` doesn't work for my project as there are a lot of old dependencies, the project contains OC and swift modules, I tried `use_frameworks! :linkage => :static`, but encounter this error after pod install. I searched and tried different solution but still failed. Can you give me some advices? Thanks.

<img width="1205" alt="image" src="https://github.com/firebase/firebase-ios-sdk/assets/66658218/0f0a8730-9cd6-470c-9e03-75707b5fec25">

**Comment by ncooke3** [comment]
Hi @haozhutw, checkout these suggestions ([option 1](https://github.com/CocoaPods/CocoaPods/issues/8607#issuecomment-514257616), [option 2](https://github.com/CocoaPods/CocoaPods/issues/8607#issuecomment-695223719)). They may help address the errors you've shared.

**Comment by gcuenca-nerdy** [comment]
> Podfiles should include either `use_frameworks!` or `use_frameworks! :linkage => :static`

Hey @paulb777 , after adding use_frameworks! in the Podfile caused many errors related to "Module map 'xxxx' not found" when building the unit tests. What can be the workaround here? Thks!

**Comment by haozhutw** [comment]
> Hi @haozhutw, checkout these suggestions ([option 1](https://github.com/CocoaPods/CocoaPods/issues/8607#issuecomment-514257616), [option 2](https://github.com/CocoaPods/CocoaPods/issues/8607#issuecomment-695223719)). They may help address the errors you've shared.

Thanks, option 1 doesn't work, with option 2 I can `pod install` successfully, but other errors occurred when I build the project. I have to revert the update, Firebase 10.23 totally breaks my project, 10.21 and below works well.

**Comment by paulb777** [comment]
Reopening to investigate making a less-disruptive migration to the Swift-implementations.

**Comment by tomerh2001** [comment]
What version has this fix?

**Comment by ncooke3** [comment]
@tomerh2001, this fix has been checked in and will be released in Firebase 10.24 (due out week of April 9).

**Comment by tomerh2001** [comment]
> @tomerh2001, this fix has been checked in and will be released in Firebase 10.24 (due out week of April 9).

Oh I see. This bug is currently preventing us from compiling our app, any temp workarounds for now? We're using:
```
    "@react-native-firebase/analytics": "^19.1.1",
    "@react-native-firebase/app": "^19.1.1",
    "@react-native-firebase/app-distribution": "^19.1.1",
    "@react-native-firebase/crashlytics": "^19.1.1",
    "@react-native-firebase/perf": "^19.1.1",
```

If I understand correctly you are the not the maintainers of these, but they do use this package under the hood - so maybe a version that previously worked (and I'll find the matching `react-native-firebase/crashlytics` version)

**Comment by EzeRangel** [comment]
@tomerh2001 

Downgrading `@react-native-firebase` to `^18.3.0` fixed for me.
